packages/ui-components/src/components/radio-button-group/radio-button-group.tsx

```diff
@@ -1,30 +1,34 @@
-import { Component, h, Prop, Host, Listen, Event, EventEmitter } from '@stencil/core';
+import { Component, h, Prop, Host, Event, EventEmitter } from '@stencil/core';
 import { IRadioButton } from '../radio-button/radio-button.types';
+import { IRadioButtonGroup, IRadioButtonGroupEvents } from './radio-button-group.types';
 
 @Component({
 	tag: 'kv-radio-button-group',
 	styleUrl: 'radio-button-group.scss',
 	shadow: true
 })
-export class KvRadioButtonGroup {
-	/** (required) List of radio buttons */
-	@Prop({ reflect: true }) buttons: { [key: string]: IRadioButton } = {};
+export class KvRadioButtonGroup implements IRadioButtonGroup, IRadioButtonGroupEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) buttons: IRadioButton[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: true }) selectedButtons?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: true }) disabled?: boolean = false;
 
-	/** When the radio button selection changes, emit the requested tab's key */
-	@Event() radioButtonChange: EventEmitter<string>;
-
-	/** Listen to custom DOM event of radio button selection */
-	@Listen('checkedChange')
-	radioChangeHandler(event: CustomEvent<string>) {
-		this.buttons[event.detail].active = !this.buttons[event.detail].active;
-		this.radioButtonChange.emit(event.detail);
-	}
+	/** @inheritdoc */
+	@Event() checkedChange: EventEmitter<string>;
 
 	render() {
 		return (
 			<Host>
-				{Object.keys(this.buttons).map(key => (
-					<kv-radio-button label={this.buttons[key].label} id={key} disabled={this.buttons[key].disabled} checked={this.buttons[key].active}></kv-radio-button>
+				{this.buttons.map(button => (
+					<kv-radio-button
+						exportparts="radio-button"
+						value={button.value}
+						label={button.label}
+						disabled={this.disabled || button.disabled}
+						checked={button.checked ?? this.selectedButtons.includes(button.value ?? button.label)}
```

Comment:
make sense to mantain the checked state inside the button configuration? If `selectedButtons` is a dictionary we don't need to make this loop to assert is checked. What you think?

Reply:
Yes, makes sense, I'll change it.